lib/grpc_kit/rpc_dispatcher.rb

```diff
@@ -28,6 +25,10 @@ def initialize(rpcs, max: DEFAULT_MAX, min: DEFAULT_MIN, interval: 30)
       @mutex = Mutex.new
 
       @min_pool_size.times { spawn_thread }
+
+      unless max == min
+        @auto_trimmer = AutoTrimmer.new(self, interval: interval).tap(&:start!)
+      end
```

Comment:
Right, the issue is AutoTrimmer starts a thread which calls into `trim` below, and that accesses `@mutex`, so `@mutex` must be initialized first.
So this fixes a general race condition, it looks good to me.